lm_eval/models/huggingface.py

```diff
@@ -441,20 +443,15 @@ def tokenizer_name(self) -> str:
     def _get_backend(
         self,
         config: Union[transformers.PretrainedConfig, transformers.AutoConfig],
-        backend: Optional[Literal["default", "causal", "seq2seq"]] = "default",
+        backend: Literal["default", "causal", "seq2seq"] = "default",
         trust_remote_code: Optional[bool] = False,
     ) -> None:
         """
         Helper method during initialization.
-        Determines the backend ("causal" (decoder-only) or "seq2seq" (encoder-decoder))
-        model type to be used.
+        Determines the backend ("causal" (decoder-only) or "seq2seq" (encoder-decoder)) model type to be used.
         sets `self.AUTO_MODEL_CLASS` appropriately if not already set.
+        Should only be called if isinstance(pretrained, str)! otherwise pass `backend` appropriately
```

Comment:
This is a change in interface, right?

I think we should still call `_get_backend` so that if someone passes an HF model we can detect backend for, it is set by us. Let's try not to force people to pass `backend` unless they absolutely have to.